Mathlib/RingTheory/Idempotents.lean

```diff
@@ -335,6 +345,143 @@ theorem CompleteOrthogonalIdempotents.of_ker_isNilpotent_of_isMulCentral
     simpa [RingHom.mem_ker, sub_eq_zero] using congr_fun h₂.symm i
   exact h₁
 
+omit [Fintype I] in
+lemma ne_sum_ortho {e : R} (he : IsPrimitiveIdempotents e)
+    (e' : I → R) (ho : OrthogonalIdempotents (R := R) e')
+    (_ : ∀ i, e' i ≠ 0) (i j : I) :
+    e ≠ e' i + e' j := by
+  have : e ≠ 0 := he.ne_zero
+  by_cases eq : i = j
+  · subst eq
+    intro rid
+    have eq : e * e = e := he.idem
+    conv_lhs at eq => rw [rid]
+    simp only [mul_add, add_mul] at eq
+    rw [ho.idem i, ← rid] at eq
+    simp only [add_right_eq_self] at eq
+    contradiction
+  · apply he.ne_sum_ortho' (e' i) (e' j)
+      ⟨fun x ↦ by
+        simp only [Nat.succ_eq_add_one, Nat.reduceAdd]
+        fin_cases x <;> simp [Nat.succ_eq_add_one, Nat.reduceAdd, Fin.zero_eta, Fin.isValue,
+            Matrix.cons_val_zero, Matrix.cons_val_one, ho.idem],
+      fun x y hxy ↦ by
+        fin_cases x <;> fin_cases y <;> simp_all [ho.2 eq]
+        exact ho.2 <| fun a ↦ eq (a.symm)⟩
+    · simp_all only [ne_eq, not_false_eq_true]
+    · simp_all only [ne_eq, not_false_eq_true]
+
+variable {ι M σ : Type*} [DecidableEq ι] [AddCommGroup M] [Module R M]
+open DirectSum
+variable (ℳ : ι → Submodule R M) [Decomposition ℳ] [(i : ι) → (x : (ℳ i)) → Decidable (x ≠ 0)]
+
+lemma decompose_unique  (rep₁ rep₂ : ⨁ i, ℳ i)
+    (h₁ : (∑ i ∈ rep₂.support, rep₂ i : M) = (∑ i ∈ rep₁.support, rep₁ i)) :
+    rep₁ = rep₂ := by
+  apply_fun (decompose ℳ).symm
+  rw [← sum_support_decompose ℳ (r := (decompose ℳ).symm rep₁),
+    ← sum_support_decompose ℳ (r := (decompose ℳ).symm rep₂)]
+  simp only [Equiv.apply_symm_apply]
+  exact h₁.symm
+
+omit [Fintype I] in
+/-- If a ring can be decomposed into direct sum of finitely many left ideals `Vᵢ`
+  where `1 = e₁ + ... + eₙ` and `eᵢ ∈ Vᵢ`, then `eᵢ` is a family of orthogonal
+  idempotents.-/
+theorem decomp_ring_ortho_idem [DecidableEq I] (V : I → Submodule R R)
```

Comment:
The name probably should be `OrthogonalIdempotents.xxx`

Reply:
Also the proof is extremely long, is it possible to break it down

Reply:
> The name probably should be `OrthogonalIdempotents.xxx`

good idea, fixed

Reply:
> Also the proof is extremely long, is it possible to break it down

fixed!